Brokerages/Backtesting/BacktestingBrokerage.cs

```diff
@@ -214,6 +215,13 @@ public void Scan()
                         continue;
                     }
 
+                    // all order fills are processed on the next bar
+                    if (order.Time == Algorithm.UtcTime)
+                    {
```

Comment:
Ok, so we keep the old behavior only for market orders. For backtesting it wouldn't have made a big difference anyway (close of current bar vs open of next bar).